HMMVoiceCreation.md

```diff
@@ -26,23 +26,115 @@ If you have already installed some of the required programs, please include thei
-* tcl supporting snack, for example [ActiveTcl](http://www.activestate.com/Products/ActiveTcl/). Note that only ActiveTcl 8.4 includes snack; 8.5+ requires manual installation.  
-* [snack](http://www.speech.kth.se/snack/download.html) library for tcl.  
-* EHMM for automatic labeling, available with [festvox-2.1](http://festvox.org/download.html)  
+**Installation instructions**
+
+All tools that require manual installation must be extracted, compiled and installed using the standard ``./configure``, ``make``, ``sudo make install`` workflow. 
+Before you start building the tools, make sure the following programs and libraries are installed on your system, since they are needed during the compilation process:
+* gcc, g++
+* libc6 (libc6-i368 if you're on a 64-bit system)
+
+In order to compile and install the patched HTK tools, first extract the [HTK-3.4.1](http://htk.eng.cam.ac.uk/ftp/software/HTK-3.4.1.tar.gz) archive as follows:
+
+    $ tar -zxvf /path/to/HTK-3.4.1.tar.gz
+    
+Then, staying in the same working directory, extract the [HDecode](http://htk.eng.cam.ac.uk/prot-docs/hdecode.shtml) archive "on top", then switch to the ``htk`` directory, which now contains the HTK and HDecode sources:
+
+    $ tar -zxvf /path/to/HDecode-3.4.1.tar.gz
+    $ cd htk
+    
+Now, extract and apply the .patch file from the [HTS-2.2_for_HTK-3.4.1](http://hts.sp.nitech.ac.jp/archives/2.2/HTS-2.2_for_HTK-3.4.1.tar.bz2) archive:
+
+    $ tar -zxvf /path/to/HTS-2.2_for_HTK-3.4.1.tar.bz2 HTS-2.2_for_HTK-3.4.1.patch
+    $ patch -p0 --strip=1 < HTS-2.2_for_HTK-3.4.1.patch
+    
+The HTK tools are now ready to be compiled and installed. It is important that all binaries produced during the build process are copied to a known location that can then either be passed as an argument to the aforementioned ``check_install_external_programs.sh`` script or simply added to the system's ``$PATH``. The easiest way to ensure this is to pass a path to the ``configure`` script using the ``--prefix`` option and making sure the ``bin`` directory under that path is contained in the system's ``$PATH`` variable. This can be achived as follows:
+
+    $ export PATH=$PATH:/path/to/external/programs/bin
+    $ ./configure --prefix=/path/to/external/programs
+    
+Once the configuration completes successfully, you can run ``make`` to compile the HTK tools. If you encounter errors during the compilation process, try re-running the ``configure`` script with the option ``--disable-hslab``, then run ``make`` again.
+
+Run ``make install`` to copy the binaries to the specified path.
+
+The procedure is very similar for the remaining tools ([SPTK-3.4.1](http://downloads.sourceforge.net/sp-tk/SPTK-3.4.1.tar.gz),[hts_engine_API-1.05](http://downloads.sourceforge.net/hts-engine/hts_engine_API-1.05.tar.gz) and EHMM): extract the archives, ``configure`` using the same path for the ``--prefix`` option, ``make`` and finally ``make install``. 
```

Comment:
EHMM doesn't use autotools, so please remove here and add a section that says just run make in the unpacked sources, then add to `PATH`